src/main/ts/ephox/photon/Reader.ts

```diff
@@ -1,12 +1,12 @@
-import { console } from '@ephox/dom-globals';
+import { console, Document, HTMLIFrameElement } from '@ephox/dom-globals';
 import { Option } from '@ephox/katamari';
-import { Element } from '@ephox/sugar';
+import { Element, Traverse } from '@ephox/sugar';
 
-var iframeDoc = function (element) {
-  var dom = element.dom();
+const iframeDoc = (element: Element<HTMLIFrameElement>): Option<Element<Document>> => {
+  const dom = element.dom();
   try {
-    var idoc = dom.contentWindow ? dom.contentWindow.document : dom.contentDocument;
-    return idoc !== undefined && idoc !== null ? Option.some(Element.fromDom(idoc)) : Option.none();
+    const idoc = dom.contentWindow ? dom.contentWindow.document : dom.contentDocument;
+    return Option.from(idoc).map(Element.fromDom);
```

Comment:
FYI there's an `Options.mapFrom`, if you prefer. Either way is fine, though.

Reply:
Oh, I didn't know that existed, that's handy thanks!